include/dlaf/eigensolver/backtransformation/mc.h

```diff
@@ -0,0 +1,200 @@
+                                                                        mat_v.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsW2(n_workspaces,
+                                                                         mat_c.distribution());
+  common::RoundRobin<matrix::Panel<Coord::Row, T, Device::CPU>> panelsT(n_workspaces, dist_t);
+
+  // Specific for V matrix layout where last column of tiles is empty
+  const SizeType num_panel_refls = mat_v.nrTiles().cols() - 1;
+
+  for (SizeType k = num_panel_refls - 1; k >= 0; --k) {
+    bool is_last = (k == num_panel_refls - 1) ? true : false;
+
+    auto& panelVV = panelsVV.nextResource();
+    auto& panelW = panelsW.nextResource();
+    auto& panelW2 = panelsW2.nextResource();
+    auto& panelT = panelsT.nextResource();
+
+    for (SizeType i = k + 1; i < mat_v.nrTiles().rows(); ++i) {
+      // Copy V panel into VV
+      auto ik = LocalTileIndex{i, k};
+
+      // Setting VV
+      if (i == k + 1) {
+        copySingleTile(mat_v.read(ik), panelVV(ik));
+        hpx::dataflow(hpx::launch::sync, unwrapping(tile::laset<T>), lapack::MatrixType::Upper, 0.f, 1.f,
+                      std::move(panelVV(ik)));
+      }
+      else {
+        hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile_v = mat_v.read(ik);
+
+        if (is_last) {
+          tile_v = splitTile(tile_v, {{0, 0}, dist_t.tileSize(GlobalTileIndex(k, k))});
+        }
+
+        panelVV.setTile(ik, mat_v.read(ik));
+      }
+
+      // Copy VV into W
+      copySingleTile(panelVV.read(ik), panelW(ik));
+    }
+
+    const GlobalTileIndex v_start{k + 1, k};
+    auto taus_panel = taus[k];
+    const SizeType taupan =
+        (is_last) ? mat_v.tileSize(GlobalTileIndex(0, m - 1)).cols() : mat_v.blockSize().cols();
+    auto kk = LocalTileIndex{k, k};
+    const LocalTileIndex diag_wp_idx{Coord::Col, k};
+
+    if (is_last) {
+      panelT.setHeight(dist_t.tileSize(GlobalTileIndex{k, k}).cols());
+      panelW.setWidth(dist_t.tileSize(GlobalTileIndex{k, k}).cols());
+    }
+
+    dlaf::factorization::internal::computeTFactor<Backend::MC>(taupan, mat_v, v_start, taus_panel,
+                                                               panelT(diag_wp_idx));
+
+    // WH = V T
+    for (SizeType i = k + 1; i < m; ++i) {
+      auto ik = LocalTileIndex{i, k};
+      hpx::shared_future<matrix::Tile<const T, Device::CPU>> tile_t = panelT.read(diag_wp_idx);
+
+      trmmPanel(executor_np, tile_t, panelW(ik));
+    }
+
+    // W2 = W C
+    for (SizeType j = 0; j < n; ++j) {
+      auto kj = LocalTileIndex{Coord::Col, j};
+      for (SizeType i = k + 1; i < m; ++i) {
+        auto ik = LocalTileIndex{i, k};
+        auto ij = LocalTileIndex{i, j};
+        panelW.setWidth(mat_v.tileSize(GlobalTileIndex{i, k}).cols());
+        gemmUpdateW2(executor_np, panelW(ik), mat_c.read(ij), (i == k + 1) ? T(0) : T(1), panelW2(kj));

```

Comment:
For more readability I would set W2 to 0 before and always sum the result.

Reply:
I will use `set0` from PR #402, if it is ok.